tests/test_structure.py

```diff
@@ -0,0 +1,88 @@
+"""Test that all services have valid structure."""
+
+from pathlib import Path
+
+import pytest
+
+
+def test_service_structure():
+    """Test that all services have the expected directory structure."""
+    repo_root = Path(__file__).parent.parent
+    services_dir = repo_root / "services"
+
+    expected_services = [
+        "clarifai-core",
+        "vault-watcher",
+        "scheduler",
+        "clarifai-ui"
+    ]
+
+    for service_name in expected_services:
+        service_dir = services_dir / service_name
+        assert service_dir.exists(), f"Service directory {service_name} not found"
+
+        # Check for pyproject.toml
+        pyproject_file = service_dir / "pyproject.toml"
+        assert pyproject_file.exists(), f"pyproject.toml not found in {service_name}"
+
+        # Check for README.md
+        readme_file = service_dir / "README.md"
+        assert readme_file.exists(), f"README.md not found in {service_name}"
+
+
+def test_service_modules_exist():
+    """Test that each service has its main module directory."""
+    repo_root = Path(__file__).parent.parent
+    services_dir = repo_root / "services"
+
+    service_modules = {
+        "clarifai-core": "clarifai_core",
+        "vault-watcher": "clarifai_vault_watcher",
+        "scheduler": "clarifai_scheduler",
+        "clarifai-ui": "clarifai_ui"
+    }
+
+    for service_name, module_name in service_modules.items():
+        service_package_dir = services_dir / service_name / module_name
+        assert service_package_dir.exists(), f"Module directory {module_name} not found in {service_name}"
+
+        # Check for __init__.py
+        init_file = service_package_dir / "__init__.py"
+        assert init_file.exists(), f"__init__.py not found in {service_name}/{module_name}"
+
+
+def test_shared_package():
+    """Test that shared package can be imported."""
+    # This test relies on the test runner (pytest) having 'shared' in its PYTHONPATH.
```

Comment:
@github: does this test work as is in our repo, or can we make changes to ease its functionality?